specs/skeleton.md

```diff
@@ -0,0 +1,99 @@
+# Skeleton
+
+## Overview
+
+A skeleton is used as a visual placeholder for an element while it is in a loading state and usually presents itself as a simplified wireframe-like version of the UI it is representing.
+
+### Background
+
+- [Github Issue #3755](https://github.com/microsoft/fast/issues/3755)
+
+### Use Cases
+
+Used to give users a sense of loading progress activity rather than merely presenting a blank page or container.
+
+### Non-goals
+
+#### Specfic Skeletons
+
+Should not be made to reflect a specific element (e.g. `skeleton-button`, `skeleton-text`, `skeleton-heading`, etc...). The skeleton should be as generic of a container as possible as to represent various elements of different shapes and sizes.
+  
+### Features
+
+- `Shape`: Circle or rect shapes can be selected. Border radius of rectagled shape will be determined by the design system.
```

Comment:
It would be possible to create most skeleton layouts with just circles and rects, though layout could be a considerable amount of effort and similarity to existing component templates. It would be cool if there were a way to apply this effect to components so they could go into that state easily.

Applying the border radius is great, and applying more aspects from the design system would help reduce this manual construction. I realize the other skeleton implementations don't follow that pattern, but they don't have an adaptive design system that could change so drastically between apps or even people. For instance, you'd represent a title with a rect, but you'd have to manually size it if there was no way to indicate it represented a page heading. Same with a button, whose size we also know based on the design system.

Maybe this is heading into v2 territory, but I'd like to see the adaptive design system play a role here and substantially reduce the markup required to create a skeleton interface for a full component or page.

Reply:
I think the feedback on how adaptive UI plays into this is good stuff to consider, but we should be hesitant to build our foundation templates and classes Specifically around adaptive UI. We are building the template for those systems as well as ours. Fast Frame and adaptive UI is just one of many design systems being supported here. If need be, we can create a more simple template and swap it out if that’s best. Foundation (And this spec) should be built to support as broad an application as we can. I’d expect for if the adaptive UI conversation to drive our implementation of this, not the foundation of it.